hwilib/descriptor.py

```diff
@@ -1,5 +1,54 @@
 import re
 
+# From: https://github.com/bitcoin/bitcoin/blob/master/src/script/descriptor.cpp
```

Comment:
why not just copy `test/functional/test_framework/descriptors.py` which I think has more review than transliteration of the C++?

Reply:
Because I didn't know that file existed. I don't think it really matters either way. At least the C++ has code comments explaining how it works.

Reply:
meh ok as long as we're generally relying on the C++ code for generation/validation we should be fine.

Reply:
Worst case let's just swap it out in a future PR.